### ISSUE TITLE
All components, including KTextBox, should have a standard focus outline

### ISSUE THREAD

**Issue by indirectlylit** [learningequality/kolibri-design-system#131]
Currently textboxes don't have a standard focus outline. For accessibility and consistency, when tabbing through the UI, _all_ keyboard-focusable elements should use the same standard outline.

Ref decision made in design system tactical:

https://www.notion.so/learningequality/we-will-try-adding-a-standard-blue-focus-outline-to-text-boxes-4f86ba39d14a4a07a9df5ea20bb717d9

Components with the issue - issue completed once all items are marked off:
- [x] KTextbox
- [x] KSwitch

**Comment by sairina** [comment]
@jtamiace and @khangmach - Is there a place where I can see what the standard blue focus outline looks like for the textboxes?

**Comment by indirectlylit** [comment]
No, the focus outline refers to the `secondary`-color border which appears around most K* components on tab-focus, but not on the text box. This outline should only appear when using tab keyboard navigation, and not when selecting UI elements with the mouse or touchpad.


![2021-02-11 17 17 37](https://user-images.githubusercontent.com/2367265/107719766-cff41300-6c8d-11eb-9dc0-8b439ab644f9.gif)

#124 is about applying a permanent background color to the text box, even in a non-focused state. 

Adding this tab focus outline will be in addition to any existing focus behaviors such as adjustments to the label and underline color/thickness.

**Comment by sairina** [comment]
@indirectlylit, @jtamiace, @khangmach This is what I have so far. Is this what we're looking for? And, also, when there's a character count (like when editing a user's full name), should the count be inside the outline?

![2021-03-03 22 41 44](https://user-images.githubusercontent.com/13563002/109922097-b8e38800-7c71-11eb-8ae0-b1e4f27a04e9.gif)

**Comment by indirectlylit** [comment]
wow, much a11y :)

And just to be clear, this does not show up when you simply click in the box with the mouse, correct?

One thing I'm seeing is that the vertical layout is shifting slightly on focus – though this might be due to the validation and not the new focus outline. Preferably, neither of these would cause a shift in height.

>  should the count be inside the outline?

I don't think it matters that much? Since validation is already inside, I might expect the count inside also.

**Comment by nucleogenesis** [comment]
@sairina is there a PR incoming for this?

**Comment by Cronus1007** [comment]
@nucleogenesis Shall I work upon this issue.

**Comment by sairina** [comment]
Hey @Cronus1007 , it's all yours! I hadn't made too much progress yet, but if you want me to push up my code I'm happy to do so - just let me know.

**Comment by Cronus1007** [comment]
@sairina Sure it will be really helpful.

**Comment by sairina** [comment]
@Cronus1007 Here you go! Thanks for your patience. https://github.com/learningequality/kolibri-design-system/pull/191

**Comment by sairina** [comment]
Unassigning myself from this issue.

**Comment by nucleogenesis** [comment]
@Cronus1007 

I've noted the two components that are affected currently - KSwitch and KTextbox

**Comment by indirectlylit** [comment]
follow-up issue: https://github.com/learningequality/kolibri-design-system/issues/257
